cmd/catalyst-node/catalyst-node.go

```diff
@@ -489,7 +510,34 @@ func parsePlaybackID(path string) (string, string, bool) {
 	// incoming path = '/hls/video+4712oox4msvs9qsf/index.m3u8'
 	// playbackID = '4712oox4msvs9qsf'
 	slice := strings.Split(m[1], "+")
-	return slice[len(slice)-1], m[2], true
+	pathTmpl := "/hls/%s/" + m[2]
+	return slice[len(slice)-1], pathTmpl, true
+}
+
+func parsePlaybackIDJS(path string) (string, string, bool) {
+	r := regexp.MustCompile(`^/json_([\w+-]+).js$`)
+	m := r.FindStringSubmatch(path)
+	if len(m) < 2 {
+		return "", "", false
+	}
+	// Incoming requests might come with some prefix attached to the
```

Comment:
I think this comment is not correct. Looks like copy-paste.

Reply:
Right - we do want the plus-splitting logic in both cases but the comment isn't quite right. I'll try and clean up both.